docs/reference/mapping/types/parent-join.asciidoc

```diff
@@ -0,0 +1,363 @@
+[[mapping-parent-join-field]]
+=== `join` datatype
+
+The `join` datatype is a special field that creates
+parent/child relation within documents of the same index.
+This field defines a set of possible relations within the documents,
+each relation being a parent name and a child name.
+A single parent/child relation can be defined as follow:
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "doc": {
+      "properties": {
+        "my_join_field": { <1>
+          "type": "join",
+          "my_parent": "my_child" <2>
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
```

Comment:
I think it'd be easier to follow if you had a complete example with a single parent and child relationship and an example with more than one. The way it is now, when I read the document I start to have questions about how to use the field and then I have to scan below the ways to configure multiple parents before I get to the "how".